src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
I'm running into a problem. The `sample_player.cpp` uses the feature_extractor code to enable doing "--report". It does not use agent.cpp's action interface, so will have no appropriate action_status state. (And the only way I've found so far - probably due to not knowing much C++ - to give the feature extractor information about the action status is to pass it (in ExtractFeatures) the `Agent` instance, so it can access getLastActionStatus (which is now an inline function, incidentally).) Any suggestions?